src/components/SearchTask.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react";
+import { useDebounce } from "use-debounce";
+import { useTasksStore } from "../stores/tasks";
+import { Task, TasksStore } from "../types/Task";
+
+export default function SearchTask() {
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const [termDebounced] = useDebounce(searchTerm, 500);
+  const allTasks = useTasksStore((state: TasksStore) => state.allTasks);
+  const updateTasks = useTasksStore((state: TasksStore) => state.updateTasks);
+  const resetTasks = useTasksStore((state: TasksStore) => state.resetTasks);
+
+  const handleSearch = (e: React.FormEvent<HTMLInputElement>) => {
+    setSearchTerm(e.currentTarget.value);
+  };
+
+  useEffect(() => {
```

Comment:
I'm wondering if a useEffect is really needed

Reply:
I'll figure it out